lib/authentication/view/edit_profile_page.dart

```diff
@@ -65,9 +128,90 @@ class _EditProfilePageState extends State<EditProfilePage> {
     );
   }
 
+  AppDialog _changeEmailConfirmationDialog(BuildContext context) {
+    final passwordController = TextEditingController();
+    return AppDialog(
+      title: S.of(context).actionConfirmChangeEmail,
+      message: S.of(context).messageChangeEmail(
+          emailController.text + S.of(context).stringEmailDomain),
+      content: [
+        TextFormField(
+          decoration: InputDecoration(
+            labelText: S.of(context).labelConfirmPassword,
+            hintText: S.of(context).hintPassword,
+          ),
+          obscureText: true,
+          controller: passwordController,
+        ),
+      ],
+      actions: [
+        AppButton(
+          key: ValueKey('change_email_button'),
+          text: S.of(context).actionConfirmChangeEmail,
+          color: Theme.of(context).accentColor,
+          width: 130,
+          onTap: () async {
+            AuthProvider authProvider =
+                Provider.of<AuthProvider>(context, listen: false);
+            if (await authProvider.verifyPassword(
+                password: passwordController.text, context: context)) {
+              if (await authProvider.changeEmail(
+                  email: emailController.text + S.of(context).stringEmailDomain,
+                  context: context)) {
+                AppToast.show(S.of(context).messageChangeEmailSuccess);
+                Navigator.pop(context, true);
+              } else {
+                AppToast.show(S.of(context).errorSomethingWentWrong);
```

Comment:
Use the _errorHandler in AuthProvider to handle the error inside the methods (look at the other usages of `catchError`).